datadog_lambda/wrapper.py

```diff
@@ -103,33 +106,40 @@ def __init__(self, func):
     def __call__(self, event, context, **kwargs):
         """Executes when the wrapped function gets called"""
         self._before(event, context)
+        response = None
         try:
-            return self.func(event, context, **kwargs)
+            response = self.func(event, context, **kwargs)
+            return response
         except Exception:
             submit_errors_metric(context)
             if self.span:
                 self.span.set_traceback()
             raise
         finally:
-            self._after(event, context)
+            self._after(event, context, response)
 
     def _before(self, event, context):
         try:
 
             set_cold_start()
             submit_invocations_metric(context)
-            # Extract Datadog trace context from incoming requests
-            dd_context = extract_dd_trace_context(event)
+            # Extract trigger tags
+            trigger_tags = extract_trigger_tags(event, context)
+            # Extract Datadog trace context and source from incoming requests
+            dd_context, trace_context_source = extract_dd_trace_context(event)
+            if trace_context_source == TraceContextSource.EVENT:
+                create_dd_metadata_subsegment(event, dd_context, trigger_tags)
```

Comment:
Actually I think the current code is problematic -- we used to only subsegment to pass the datadog trace context when the incoming http event carries a Datadog trace context, but I think you actually need to **always** create the subsegment whenever you can extract trigger tags from the event?

Reply:
Ah that's right. My thinking was that we only needed to add trigger tags when hybrid tracing was in use. I updated this with a function to add generic tags into the `root_span_metadata` field.